### ISSUE TITLE
Using the "new" toolbar manager still triggers warnings referring to V. 1.5 / 2.1

### ISSUE THREAD

**Issue by EBenkler** [matplotlib/matplotlib#15284]
### Bug report

**Bug summary**

Using the "new" toolbar manager still triggers warnings referring to V. 1.5 / 2.1, although we are now at V.3.1.1. I wonder if the warning still applies, or if it could be removed, since it is rather annoying and is triggered e.g. in plt.figure()? Are there any workarounds? As far as I understand, there is an old and a new toolbar manager, but I don't know how I could implement a toggle button using the old one.

**Code for reproduction**

```python
import numpy as np
import matplotlib.pyplot as plt
plt.rcParams['toolbar'] = 'toolmanager'
from matplotlib.backend_tools import ToolToggleBase

class toggle_example(ToolToggleBase):
    default_keymap = 't'
    description = 'Toggle Button'

t= np.arange(1e5)/1e5
y= np.random.randn(int(1e5))
    
plt.close(fig=1)
plt.figure(1)
plt.plot(t,y)

plt.gcf().canvas.manager.toolmanager.add_tool('Toggle example', toggle_example)
plt.gcf().canvas.manager.toolbar.add_tool('Toggle example', 'zoompan', 2)
```  
**Actual outcome**

```
Treat the new Tool classes introduced in v1.5 as experimental for now, the API will likely change in version 2.1 and perhaps the rcParam as well
__main__:15: UserWarning: The new Tool classes introduced in v1.5 are experimental; their API (including names) will likely change in future versions.
__main__:18: UserWarning: The new Tool classes introduced in v1.5 are experimental; their API (including names) will likely change in future versions.
```

**Expected outcome**

I would expect not to see these warnings each time a new figure is opened.

**Matplotlib version**

  * Operating system: Windows 10
  * Matplotlib version: 3.1.1
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.6.7
  * Jupyter version (if applicable):
  * Other libraries: 
conda default channel installed on Miniconda

**Comment by anntzer** [comment]
This is because there are still disagreements about the new toolbar API.
You can suppress the warning with https://docs.python.org/3/library/warnings.html#overriding-the-default-filter.

**Comment by anntzer** [comment]
Actually I agree that the warning is just too annoying to suppress right now, no need to warn again and again.  I guess we could move it to the rcParam validator instead with the patch below (feel free to pick it up as a PR); after that patch you'd just need to wrap the `rcParams["toolbar"] = "toolmanager"` line with the warnings contextmanager.
```patch
diff --git i/lib/matplotlib/backend_managers.py w/lib/matplotlib/backend_managers.py
index ce9809aad..a6b0e0361 100644
--- i/lib/matplotlib/backend_managers.py
+++ w/lib/matplotlib/backend_managers.py
@@ -1,12 +1,8 @@
-import logging
-
 import matplotlib.cbook as cbook
 import matplotlib.widgets as widgets
 from matplotlib.rcsetup import validate_stringlist
 import matplotlib.backend_tools as tools
 
-_log = logging.getLogger(__name__)
-
 
 class ToolEvent:
     """Event for tool manipulation (add/remove)."""
@@ -51,10 +47,6 @@ class ToolManager:
     """
 
     def __init__(self, figure=None):
-        _log.warning('Treat the new Tool classes introduced in v1.5 as '
-                     'experimental for now, the API will likely change in '
-                     'version 2.1 and perhaps the rcParam as well')
-
         self._key_press_handler_id = None
 
         self._tools = {}
diff --git i/lib/matplotlib/backend_tools.py w/lib/matplotlib/backend_tools.py
index d4dbf321f..1898f4037 100644
--- i/lib/matplotlib/backend_tools.py
+++ w/lib/matplotlib/backend_tools.py
@@ -80,9 +80,6 @@ class ToolBase:
     """
 
     def __init__(self, toolmanager, name):
-        cbook._warn_external(
-            'The new Tool classes introduced in v1.5 are experimental; their '
-            'API (including names) will likely change in future versions.')
         self._name = name
         self._toolmanager = toolmanager
         self._figure = None
diff --git i/lib/matplotlib/rcsetup.py w/lib/matplotlib/rcsetup.py
index ad807d72a..1dde30244 100644
--- i/lib/matplotlib/rcsetup.py
+++ w/lib/matplotlib/rcsetup.py
@@ -310,6 +310,17 @@ validate_toolbar = ValidateInStrings(
     _deprecated_since="3.3")
 
 
+def _validate_toolbar(s):
+    s = ValidateInStrings(
+        'toolbar', ['None', 'toolbar2', 'toolmanager'], ignorecase=True)(s)
+    if s == 'toolmanager':
+        cbook._warn_external(
+            "Treat the new Tool classes introduced in v1.5 as experimental "
+            "for now, the API will likely change in version 2.1 and perhaps "
+            "the rcParam as well.")
+    return s
+
+
 @cbook.deprecated("3.3")
 def _make_nseq_validator(cls, n=None, allow_none=False):
 
@@ -1008,7 +1019,7 @@ def _convert_validator_spec(key, conv):
 _validators = {
     "backend":           validate_backend,
     "backend_fallback":  validate_bool,
-    "toolbar":           _ignorecase(["none", "toolbar2", "toolmanager"]),
+    "toolbar":           _validate_toolbar,
     "datapath":          validate_any,  # see _get_data_path_cached
     "interactive":       validate_bool,
     "timezone":          validate_string,
```

**Comment by brobr** [comment]
Hi, thanks for picking this up again.

I copied your patch text to a textfile 'toolmanager_warn_patch.txt' (witrh extra hard return/new line at the end) and used that to patch the source before building:
`patch --verbose -p1 < $CWD/toolmanager_warn_patch.txt`
This worked as far as I can see:
<pre> 
matplotlib-3.3.1/versioneer.py
Hmm...  Looks like a unified diff to me...
The text leading up to this was:
--------------------------
|diff --git i/lib/matplotlib/backend_managers.py w/lib/matplotlib/backend_managers.py
|index ce9809aad..a6b0e0361 100644
|--- i/lib/matplotlib/backend_managers.py
|+++ w/lib/matplotlib/backend_managers.py
--------------------------
patching file lib/matplotlib/backend_managers.py
Using Plan A...
Hunk #1 succeeded at 1.
Hunk #2 succeeded at 47.
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|diff --git i/lib/matplotlib/backend_tools.py w/lib/matplotlib/backend_tools.py
|index d4dbf321f..1898f4037 100644
|--- i/lib/matplotlib/backend_tools.py
|+++ w/lib/matplotlib/backend_tools.py
--------------------------
patching file lib/matplotlib/backend_tools.py
Using Plan A...
Hunk #1 succeeded at 80.
Hmm...  The next patch looks like a unified diff to me...
The text leading up to this was:
--------------------------
|diff --git i/lib/matplotlib/rcsetup.py w/lib/matplotlib/rcsetup.py
|index ad807d72a..1dde30244 100644
|--- i/lib/matplotlib/rcsetup.py
|+++ w/lib/matplotlib/rcsetup.py
--------------------------
patching file lib/matplotlib/rcsetup.py
Using Plan A...
Hunk #1 succeeded at 301 (offset -9 lines).
Hunk #2 succeeded at 1067 (offset 48 lines).
done
</pre>


The built matplotlib can be imported into a python shell but not the pyplot module

<pre>
import matplotlib

import matplotlib.pyplot
---------------------------------------------------------------------------
ImportError                               Traceback (most recent call last)
<ipython-input-6-864e826dab68> in <module>
----> 1 import matplotlib.pyplot

/usr/lib64/python3.8/site-packages/matplotlib/pyplot.py in <module>
   2318     dict.__setitem__(rcParams, "backend", rcsetup._auto_backend_sentinel)
   2319 # Set up the backend.
-> 2320 switch_backend(rcParams["backend"])
   2321 
   2322 # Just to be safe.  Interactive mode can be turned on without

/usr/lib64/python3.8/site-packages/matplotlib/pyplot.py in switch_backend(newbackend)
    258     backend_name = cbook._backend_module_name(newbackend)
    259 
--> 260     class backend_mod(matplotlib.backend_bases._Backend):
    261         locals().update(vars(importlib.import_module(backend_name)))
    262 

/usr/lib64/python3.8/site-packages/matplotlib/pyplot.py in backend_mod()
    259 
    260     class backend_mod(matplotlib.backend_bases._Backend):
--> 261         locals().update(vars(importlib.import_module(backend_name)))
    262 
    263     required_framework = _get_required_interactive_framework(backend_mod)

/usr/lib64/python3.8/importlib/__init__.py in import_module(name, package)
    125                 break
    126             level += 1
--> 127     return _bootstrap._gcd_import(name[level:], package, level)
    128 
    129 

/usr/lib64/python3.8/site-packages/ipykernel/pylab/backend_inline.py in <module>
      7 
      8 import matplotlib
----> 9 from matplotlib.backends.backend_agg import (
     10     new_figure_manager,
     11     FigureCanvasAgg,

/usr/lib64/python3.8/site-packages/matplotlib/backends/backend_agg.py in <module>
     47 from matplotlib.path import Path
     48 from matplotlib.transforms import Bbox, BboxBase
---> 49 from matplotlib.backends._backend_agg import RendererAgg as _RendererAgg
     50 
     51 

ImportError: /usr/lib64/python3.8/site-packages/matplotlib/backends/_backend_agg.cpython-38-x86_64-linux-gnu.so: file too short

</pre>

I cannot immediately see what could have been wrong at my end...

**Comment by anntzer** [comment]
Closed by #19461.
